### PR TITLE
feat: better call graph errors

### PR DESCRIPTION
**Pull request by darmalovan**
- [x] Ready for review
- [ ] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Present more user-friendly messages for call graph errors
This relies on https://github.com/snyk/java-call-graph-builder/pull/11

Also, moved reachable vulns system tests to its own file.

#### What are the relevant tickets?
https://snyksec.atlassian.net/browse/FLOW-202

#### Screenshots
image

image

### CODE REVIEW

File: lib/error-format.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+export function formatCallGraphError(error: Error): string {
+  if (error.message === 'Could not find target folder') {
+    return 'Couldn\'t scan for reachable vulns, to do so your code should be compiled, please run `mvn compile`.';
```
Comment (dkontorovskyy) [inline lib/error-format.ts:3]: Failed to scan for reachable vulns. Compile your code by executing `mvn compile` and try again, please.

What you think of this option?

File: lib/error-format.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  }
+  if (error.message === 'No entrypoints found') {
+    return 'Couldn\'t scan for reachable vulns, didn\'t find any entry point to your application.';
```
Comment (dkontorovskyy) [inline lib/error-format.ts:6]: Failed to scan fro reachable vulns. Application's entry point has not been found.

File: lib/error-format.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  }
+  return 'Couldn\'t scan for reachable vulns, please contact our support or submit an issue at https://github.com/snyk/java-call-graph-builder/issues.';
```
Comment (dkontorovskyy) [inline lib/error-format.ts:8]: Failed to scan for reachable vulns. Please contact our support or submit an issue at `https://github.com/snyk/java-call-graph-builder/issues.'

File: lib/error-format.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+export function formatGenericPluginError(
+  error: Error,
+  mavenCommand: string,
+  mvnArgs: string[],
+): string {
+  const mavenArguments: string = mvnArgs.join(' ');
```
Comment (dkontorovskyy) [inline lib/error-format.ts:16]: const fullCommand = [mavenCommand, ... mavenArguaments].join(' ');

File: lib/index.ts [hunk 5]
```diff
@@ -98,10 +100,15 @@ export async function inspect(
...
-      );
-      debug('got call graph successfully');
+      try {
+        callGraph = await javaCallGraphBuilder.getCallGraphMvn(
+          path.dirname(targetPath),
+        );
+        debug('got call graph successfully');
+      } catch (err) {
+        debug('call graph error: ', err);
+        throw new CallGraphError(err.message, err);
```
Comment (dkontorovskyy) [inline lib/index.ts:110]: `throw new CallGraphError(err);`
and inside CallGraphError distruct message from err

File: lib/error-format.ts (hunk 2 above)
Comment (muscar) [reply lib/error-format.ts:6]: How about: "Failed to scan for reachable vulns. Couldn't find the application entry point"?

File: lib/error-format.ts (hunk 1 above)
Comment (muscar) [reply lib/error-format.ts:3]: How about: "Failed to scan for reachable vulns. Please compile your code by running `mvn compile`"?

File: lib/index.ts (hunk 5 above)
Comment (darmalovan) [reply lib/index.ts:110]: This is an implementation detail that I don't want to enforce currently in this class.

File: lib/error-format.ts (hunk 1 above)
Comment (darmalovan) [reply lib/error-format.ts:3]: Thanks, I'm combining both of your suggestions

**Comment by darmalovan** [comment]
@dkontorovskyy @muscar thanks for the comments, I addressed ready for another review.

**Pull request merged**